main/src/ca/uwaterloo/flix/language/phase/unification/ClassEnvironment.scala

```diff
@@ -17,31 +17,32 @@
 package ca.uwaterloo.flix.language.phase.unification
 
 import ca.uwaterloo.flix.api.Flix
-import ca.uwaterloo.flix.language.ast.{ResolvedAst, Symbol, Type, TypedAst}
-import ca.uwaterloo.flix.util.{InternalCompilerException, Result}
+import ca.uwaterloo.flix.language.ast.{Symbol, Type, TypedAst}
 import ca.uwaterloo.flix.util.Result.{ToErr, ToOk}
 import ca.uwaterloo.flix.util.collection.MultiMap
+import ca.uwaterloo.flix.util.{InternalCompilerException, Result}
 
 import scala.annotation.tailrec
 
-object ContextReduction {
+case class ClassEnvironment(instances: MultiMap[Symbol.ClassSym, ClassEnvironment.Instance]) {
```

Comment:
I think this definition should either live in `TypedAst` or alternatively in the `Ast.scala` file.

Reply:
And then the functions themselves would each have a ClassEnvironment parameter?

What would you propose calling this object?

Reply:
Oh I see. 

Yeah, in general I try to avoid "OO"-style programming. For example, you will notice that most AST nodes have no methods defined on them whatsoever. So yeah, you need extra parameters.